File: Week8/StretchProblems/Objc - find numbers divisible by x/Objc - find numbers divisible by x/ViewController.m
//
//  ViewController.m
//  Objc - find numbers divisible by x
//
//  Created by Kaleo Kim on 3/25/16.
//  Copyright © 2016 Kaleo Kim. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    [self devisibleByX:4 y:12];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)devisibleByX:(int)x :(int)y {
    
    for (int i = 0; i < y; i++) {
        if (i % x == 0) {
            NSLog(@"%d, ", i);
        }
    }
    
}

@end

File: Review/Unit 3/MiniProject/DeckOfCard- Mini Poject/DeckOfCard- Mini Poject/ViewController.swift
//
//  ViewController.swift
//  DeckOfCard- Mini Poject
//
//  Created by Kaleo Kim on 4/18/16.
//  Copyright © 2016 Kaleo Kim. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet var cardImageView: UIImageView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func drawCardTapped(sender: AnyObject) {
        
        CardController.drawCard { (card) in
            if let card = card {
                
                ImageController.imageForURL(card.imageEndpoint, completion: { (image) in
                    if let image = image {
                        dispatch_async(dispatch_get_main_queue(), { () -> Void in
                            self.cardImageView.image = image
                        })
                    }
                })
                
            }
        }

        
    }
    

}

